modules/gapi/perf/streaming/gapi_streaming_source_perf_tests.cpp

```diff
@@ -0,0 +1,95 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifdef HAVE_ONEVPL
+
+#include "../perf_precomp.hpp"
+#include "../../test/common/gapi_tests_common.hpp"
+#include <opencv2/gapi/streaming/onevpl/source.hpp>
+#include <opencv2/gapi/streaming/cap.hpp>
+
+namespace opencv_test
+{
+using namespace perf;
+
+const std::string files[] = {
+    "highgui/video/big_buck_bunny.h265",
+    "highgui/video/big_buck_bunny.h264",
+};
+
+const std::string codec[] = {
+    "MFX_CODEC_HEVC",
+    "MFX_CODEC_AVC"
+};
+
+using source_t = std::string;
+using codec_t = std::string;
+using source_description_t = std::tuple<source_t, codec_t>;
+
+class OneVPLSourcePerfTest : public TestPerfParams<source_description_t> {};
+class VideoCapSourcePerfTest : public TestPerfParams<source_t> {};
+
+PERF_TEST_P_(OneVPLSourcePerfTest, TestPerformance)
+{
+    if (!initTestDataPathSilent()) {
+        throw SkipTestException("env variable OPENCV_TEST_DATA_PATH was not configured");
+    }
+    using namespace cv::gapi::wip::onevpl;
+
+    const auto params = GetParam();
+    source_t src = findDataFile(get<0>(params), false);
+    codec_t type = get<1>(params);
+
+    std::vector<CfgParam> cfg_params {
+        CfgParam::create<std::string>("mfxImplDescription.Impl", "MFX_IMPL_TYPE_HARDWARE"),
+        CfgParam::create("mfxImplDescription.mfxDecoderDescription.decoder.CodecID", type),
+    };
+
+    cv::gapi::wip::IStreamSource::Ptr source_ptr;
+    try {
+        source_ptr = cv::gapi::wip::make_onevpl_src(src, cfg_params);
+    } catch(...) {
+        throw SkipTestException("G-API compiled without `WITH_GAPI_ONEVPL=ON`");
+    }
+
+    cv::gapi::wip::Data out;
+    TEST_CYCLE()
+    {
+        source_ptr->pull(out);
+    }
```

Comment:
> Wrapping TEST_CYCLE for all case and to incorporate a source creation also will measure MFX library/DX11 CoCreate load/unload and so on and doesn't give me relative result performance in comparison with VideoCapdecode api

This makes sense, but can it be done only once before the `TEST_CYCLE()` body kicks in? Is there a way to decompose it to prologue/epilogue so that initialization/deinitialization for every individual run is left out of profile?